analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/HashesShouldHaveUnpredictableSalt.cs

```diff
@@ -26,5 +26,27 @@ public sealed class HashesShouldHaveUnpredictableSalt : HashesShouldHaveUnpredic
 
     protected override DiagnosticDescriptor Rule => S2053;
 
-    public override bool ShouldExecute() => true;
+    public override bool ShouldExecute()
+    {
+        var walker = new VulnerableTypeWalker();
+        walker.SafeVisit(Node);
+        return walker.Result;
+    }
+
+    private sealed class VulnerableTypeWalker : SafeCSharpSyntaxWalker
+    {
+        public bool Result { get; private set; }
+
+        public override void Visit(SyntaxNode node)
+        {
+            if (node is ObjectCreationExpressionSyntax || ImplicitObjectCreationExpressionSyntaxWrapper.IsInstance(node))
+            {
+                Result = true;
+            }
+            else
+            {
+                base.Visit(node);
+            }
+        }
```

Comment:
There's an explicit visitor for object creation:
```suggestion
        public override void Visit(SyntaxNode node)
        {
            if (!Result)
            {
                base.Visit(node);
            }
        }

        public override void VisitObjectCreationExpression(ObjectCreationExpressionSyntax node) =>
            Result = true;
```

Reply:
We're using a lower version of Roslyn, so there's no `VisitXXX` method `ImplicitObjectCreation`.

Reply:
I've moved the `ObjectCreation` to a separate `Visit` method and left `ImplicitObjectCreation` in the default `Visit` method.